Otherworldly Pines/Assets/Scripts/FallingDamage.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FallingDamage : MonoBehaviour
+{
+    public float damageImpulseThreshold = 5000f;
+    public float highestDamagingImpulse = 10000f;
+    public float lowestDamageDealt = 10f;
+    public float HighestDamageDealt = 30f;
+
+    private Vector3 objectVelocity;
+
+    // Update is called once per frame
+    void Update()
+    {
+        objectVelocity = this.GetComponent<Rigidbody2D>().velocity;
+
+        if (objectVelocity.y > damageImpulseThreshold)
+        {
+
+        }
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        float collisionForce = GetImpactForce(collision);
+        Debug.Log(collisionForce);
+        if (collision.gameObject.tag == "Player")
+        { 
+            collision.gameObject.GetComponent<PlayerHealth>().TakeDamage(ProportionalDamageDeal(collisionForce));
+        }
+    }
+
+    private float ProportionalDamageDeal(float impulse)
+    {
+        if (impulse < damageImpulseThreshold)
+            return 0f;
+        if (impulse < highestDamagingImpulse) // objectVelocity is big enough to deal damage
+            return (impulse / highestDamagingImpulse) * HighestDamageDealt;
+        else
+            return HighestDamageDealt;
+
+    }
+
+    private float GetImpactForce(Collision2D collision)
+    {
+        float impulse = 0F;
+
+        foreach (ContactPoint2D point in collision.contacts)
+        {
+            impulse += point.normalImpulse;
+        }
+
+        return impulse / Time.fixedDeltaTime;
```

Comment:
Why divide by time?

Reply:
Not sure either lol. Got this calculation from the web. Since collision2d doesn't have an impulse built into the class, I gotta find some way to mimic it, and this is someone else's answer for the work-around.